src/libraries/Silk.Net.Windowing/Abstract/Version32.cs

```diff
@@ -0,0 +1,70 @@
+﻿
+using System;
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+namespace Silk.NET.Core
+{
+    /// <summary>
+    /// A 32-bit version structure.
+    /// </summary>
+    public readonly struct Version32
+    {
+        /// <summary>
+        /// The underlying Vulkan-compatible 32-bit version integer.
+        /// </summary>
+        public uint Value { get; }
+        /// <summary>
+        /// Creates a Vulkan version structure from the given major, minor, and patch values.
+        /// </summary>
+        /// <param name="major">The major value.</param>
+        /// <param name="minor">The minor value.</param>
+        /// <param name="patch">The patch value.</param>
+        public Version32(uint major, uint minor, uint patch) => Value = major << 22 | minor << 12 | patch;
```

Comment:
Shouldn't the patch and minor be masked to the appropriate width? Like this patch and minor can bleed into major

Reply:
This is the current implementation sitting in 2.X, that being said I can fix that.

Reply:
Yeah the 2.x implementation might also have that fault, I've just never caught it there

Reply:
I will fix when I get home and no, it DOES have that fault, it was copy pasted from 2.X